modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/itemselector/plugin.js

```diff
@@ -51,6 +51,30 @@
 			}
 		},
 
+		_checkImageWidth(editor, editorContent, imageSrc) {
+			if (imageSrc.url) {
+				imageSrc = imageSrc.url;
+			}
+
+			if (
+				!editor.window.$.AlloyEditor &&
+				!editorContent.id.endsWith('BalloonEditor')
+			) {
+				editorContent =
+					editorContent.querySelector('iframe').contentDocument;
+			}
+
+			const imgElement = editorContent.querySelector(
+				`img[src='${imageSrc}']`
+			);
+
+			imgElement.onload = function () {
+				if (this.width === 0) {
+					this.setAttribute('width', '150px');
+				}
+			};
```

Comment:
I'm aware this is very hacky and I tried other approaches but I couldn't find a better way to do it. I'm more than opened to listen to other options.

It's working fine, I don't find it glitchy. This is only affecting to images that are not rendered and after passing through that block, they got a width assigned and then they get rendered. I mean, final user is not seeing how something is re-rendered, it is just rendered a little tiny bit later than regular images. It is not noticiable for final user unless they compare loading times for different images, in which case I guess they might found a different of a few miliseconds when the original image has no width. That's all.

Originally, I tried setting this in the code injecting HTML (in both, CKEditor plugin and Adaptive Media) but at that point I couldn't check the actual width of the image. It is not until it's on DOM when I can actually see whether that width is 0 or not.

If I'm honest, I regret we accepted as a valid case the idea of rendering images with no width. We could avoid this hacky code, and as of today, it is even not actually allowed by _out-of-the-box_ CKEditor 4. But since we admitted this as a bug and not as a product limitation in a previous version, it could be seen as a regression bug if we remove this "feature" now.

Reply:
OK, thanks for clarification! If it's just a tiny time delay, it's no problem. If the issue reoccurs for CKEditor 5, we can give more effort on overhauling the whole solution.